lib/diagnostic.rb

```diff
@@ -7,29 +7,52 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
-  end
+
+  include Comparable
+
+    def <=>(list)
+      [RANKS.index(rank), SUITS.index(suit)] <=>
+        [RANKS.index(other.rank), SUITS.index(other.suit)]
+
+
+  #   include Enumerable
+  #
+  #   def each
+  #     @list = list
+  #     current_list_item = 0
+  #
+  #     while current_list_item < list.length
+  #
+  #       if current_list_item[i] < current_list_item[i+1]
+  #         yield current_list_item
+  #         current_list_item += 1
+  #       end
+  #     end
+  # end
```

Comment:
Whoa got a lot going on for this first one.  You probably overthought it, here's a quick way we can do it:

`list.max_by(&:length)`